astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -32,9 +32,18 @@ def cirs_to_altaz(cirs_coo, altaz_frame):
     # we use the same obstime everywhere now that we know they're the same
     obstime = cirs_coo.obstime
 
-    usrepr = cirs_coo.represent_as(UnitSphericalRepresentation)
-    cirs_ra = usrepr.lon.to(u.radian).value
-    cirs_dec = usrepr.lat.to(u.radian).value
+    if cirs_coo.distance.unit is u.one:
```

Comment:
Good point.  In nearly all practical cases it should exist because the default representation is `Spherical`, *but* the following produces an error:
```
c = CIRS(ra=..., dec=...)
c.representation = UnitSphericalRepresentation
c.transform_to(AltAz(...))
```
because forcing the representation to unit removes the distance.  Will at least make sure it's internally consistent here, but you're right that we should make that all more consistent (either a method or a consistent one-liner idiom that's recommended in the docs)